scsd-cdh/OBC
# Read Voltage Function complete

Added a Read_Voltage function. Function converts ADC value to mV.

## Description (tell us what you changes or added or removed please)
Read_Voltage function is included in the ADC_READ code. Code reads the ADC value
and converts it to mV.

## How to test (if any tests done)
1. Open a file explorer and navigate to wherever OBC project copy is saved. (Refer as File Explorer 1) 
2. Open a second explorer and navigate to the same directory as OBC project. (Refer as File Explorer 2)
3. On File Explorer 2, add new folder and save the path
4. Open CCS C Compiler and Press Create New Project
5. Name your Project
6. Select MSP430FR5969 as your board
7. Select Empty Project with main.c and press finish
8. On the Project Explorer, you should find _main.c_ . **Right-click** it and select:
    _**Show In > System**_ Explorer. (Refer as File Explorer 3)
9. Close File Explorer 2.
10. On File Explorer 1, Navigate to : .../OBC/DRIVERS/MSP430
11. Copy the ADC and GPIO folder from File Explorer 1 and Paste onto File Explorer 3. 
12. Close File Explorer 1 and File Explorer 3
13. On main.c, copy and paste the test code attached here :  [main.txt](https://github.com/user-attachments/files/18189518/main.txt)
14. On the Project Explorer, Right-CLick and Select Properties
15. Navigate to **_Build > MSP430 Compiler >_ Include Options**
16. In the Include Options table, add the folders shown in the picture
![image](https://github.com/user-attachments/assets/729a568e-3e84-4700-a534-546de057000d)
17. If the ADC and GPIO folders do not automatically appear on CCS' project explorer, then delete the **Debug** folder
18. Press CTLR + B to build project.
19. Flash the code
20. Go to ADC > ADC_READ.c and set a breakpoint at line 61.
21. Press the Debug button.
22. Press Run until breakpoint
23. Press Step and it will give you the voltage to return
24. Plug VCC to P1.4 and repeat steps 19 to 21


Closes : #13 

## Checklist
- [x] Did you add any comments
- [x] Did you test it at all ?! Good Boy/Girl
